src/vs/workbench/services/search/common/queryBuilder.ts

```diff
@@ -240,6 +240,9 @@ export class QueryBuilder {
 		const extraFileResources = options.extraFileResources && options.extraFileResources.filter(extraFile => pathIncludedInQuery(queryProps, extraFile.fsPath));
 		queryProps.extraFileResources = extraFileResources && extraFileResources.length ? extraFileResources : undefined;
 
+		const threads = this.configurationService.getValue<number>('search.threads') ?? undefined;
```

Comment:
Since this is similar to `disregardIgnoreFiles` in that it's a setting that drives the query, I'd probably prefer this to be read into the configuration service by the caller and integrated into the `ICommonQueryBuilderOptions`

Reply:
It seems these are the callers of `queryBuilder`: `anyThingQuickAccess`, `notebookSearchService`, `textSearchQuickAccess`, `searchView`, `searchEditor`, `terminalLinkOpener`, `workspaceContains`. 

I'm not quite sure how to add the configuration to all callers without duplicating the code:

```ts
// searchView.ts
const options: ITextQueryBuilderOptions = {
	isSmartCase: this.searchConfig.smartCase,
	expandPatterns: true, 
    threads: this.searchConfig.threads, // add it here
};

// searchEditor.ts
const options: ITextQueryBuilderOptions = {
	_reason: 'searchEditor',
    threads: this.searchConfig.threads, // add it here
};

/* similar changes 5 more times */
```

It seems simpler to read the setting once in `queryBuilder`. Perhaps I am missing something?

Reply:
Hmm. Perhaps it makes sense for now to just call it from the the queryBuilder. But perhaps this line could benefit from following the same way of getting the search configuration (`this.configurationService.getValue<ISearchConfiguration>`) as the rest of the queryBuilder.

Reply:
That's changed already. :)